contributors/design-proposals/pod-priority-resourcequota.md

```diff
@@ -0,0 +1,162 @@
+    * [Expected behavior of ResourceQuota admission controller](#resourcequota-admission-controller)
+
+
+## Objective
+
+This feature is designed to make `ResourceQuota` become priority aware, several sub-tasks are included.
+
+1. Expand `Scopes` in `ResourceQuotaSpec` to represent priority class names and corresponding behavior.
+2. Incorporate corresponding behavior in quota checking process.
+3. Update the `ResourceQuota` admission controller to check priority class name and perform expected admission.
+
+### Non-Goals 
+
+* Add priority in Pod spec (this is implemented separately in: [45610](https://github.com/kubernetes/kubernetes/pull/45610))
+
+## Background 
+
+Since we already have [priority field in Pod spec](https://github.com/kubernetes/kubernetes/pull/45610), 
+Pods can now be classified into different priority classes. We would like to be able to create quota for various priority classes in order to manage cluster resources better and limit abuse scenarios. 
+
+One approach to implement this is adding priority class name field to `ResourceQuota` API definition. While this arbitrary field of API object will introduce inflexibility to potential change in future and also, not adequate to express all semantics.
+
+Thus, we decide to reuse the existing `Scopes` of `ResourceQuotaSpec` to provide a richer semantics for quota to cooperate with priority classes. 
+
+## Overview 
+
+This design doc introduces how to define a group of priority class scopes for the quota to match with, and explains how quota enforcement logic is changed to apply the quota to pods with the given priority classes.
+
+## Detailed Design 
+
+### Expected behavior of ResourceQuota
+
+Four `Scopes` will be added to `ResourceQuota` to describe its relationship with priority class name:
+
+| Scope | Description |
+| ----- | ----------- |
+| PriorityClassNameIn:priorityName1:priorityName2 | Match `kind=Pod` where `spec.priorityClassName in (priorityName1, priorityName2)` |
+| PriorityClassNameNotIn:priorityName1:priorityName2 | Match `kind=Pod` where `spec.priorityClassName notin (priorityName1, priorityName2)` |
+| PriorityClassNameExists |  Match `kind=Pod` where `spec.priorityClassName` |
+| PriorityClassNameNotExists |  Match `kind=Pod` where `!spec.priorityClassName` |
+
+*Example: Match all pods independent of priority*
+
+`$ kubectl create quota example --hard=count/pods=10`
+
+*Example: Match all pods where priority is foo*
+
+`$ kubectl create quota example --hard=count/pods=10 --scopes=PriorityClassNameIn:foo`
+
+*Example: Match all pods where there is no priority*
+
+`$ kubectl create quota example --hard=count/pods=10 --scopes=PriorityClassNameNotExists`
+
+To restrict usage of a particular priority without explicit quota, we can do the following:
+
+*Example: Any pod with priority cluster-services, or other-important-stuff must require explicit quota*
+
+```yaml
+apiVersion: apiserver.k8s.io/v1alpha1
+kind: AdmissionConfiguration
```

Comment:
These examples should go in the next section, Controller changes.
Here examples should be for `ResourceQuota` objects.